test/logger.ts

```diff
@@ -14,26 +14,35 @@
  * limitations under the License.
  */
 
-import {Logger, logger} from '@google-cloud/common';
+import {Logger, logger, LoggerConfig} from '@google-cloud/common';
 
 const PASS_THROUGH_LOG_LEVEL = Number(process.env.GCLOUD_TEST_LOG_LEVEL || 0);
+// Capture the value of common.Logger so that we don't enter an infinite loop
+// if common.Logger is wrapped elsewhere.
+// tslint:disable-next-line:variable-name
+const OriginalLogger = Logger;
 
 // tslint:disable-next-line:no-any
-type LoggerFunction = (message: any, ...args: any[]) => void;
+type LoggerFunction<R> = (message: any, ...args: any[]) => R;
 
-export class TestLogger implements Logger {
-  private logs:
-      {[k in keyof Logger]:
-           string[]} = {error: [], warn: [], info: [], debug: [], silly: []};
-  private innerLogger = logger({level: logger.LEVELS[PASS_THROUGH_LOG_LEVEL]});
+export class TestLogger extends Logger {
+  private logs: {[k in keyof Logger]: string[]} =
+      {silent: [], error: [], warn: [], info: [], debug: [], silly: []};
+  private innerLogger =
+      new OriginalLogger({level: logger.LEVELS[PASS_THROUGH_LOG_LEVEL]});
 
-  private makeLoggerFn(logLevel: keyof Logger): LoggerFunction {
+  constructor(options?: Partial<LoggerConfig>) {
```

Comment:
Why do you need this ctor when you only calls `super(options)`?

Reply:
TSC will compile an implicit constructor to an explicit one using spread args if compiling to ES6.

Reply:
Really? `Logger`'s ctor looks exactly the same as this one: https://github.com/googleapis/nodejs-common/blob/master/src/logger.ts#L70

TS still uses spread args? And why does it matter with this test?

Reply:
It's all about Node 4 compatibility...

I think nodejs-common compiles to ES5. We can't do that since we use generators in some tests, so we just don't use ES6 features that aren't supported in Node 4.